src/FacebookEventForwarder.js

```diff
@@ -298,6 +310,70 @@
                 return !isNaN(parseFloat(n)) && isFinite(n);
             }
 
+            function parseMappings(str) {
+                if (!str) return [];
+                try { return JSON.parse((str + '').replace(/&quot;/g, '"')); } catch (_) { return []; }
+            }
+
+            // Resolve a simple product key by checking top-level first, then Attributes bag
+            function getField(product, key) {
+                if (!product || !key) return undefined;
+                if (Object.prototype.hasOwnProperty.call(product, key)) {
+                    return product[key];
+                }
+                var attrs = product.Attributes;
+                if (attrs && typeof attrs === 'object' && Object.prototype.hasOwnProperty.call(attrs, key)) {
+                    return attrs[key];
+                }
+                return undefined;
+            }
+
+            function buildProductContents(productList) {
+                if (!productList || !productList.length) {
+                    return [];
+                }
+                try {
+                    var setting = self._settings;
+                    var mappings = parseMappings(setting.productAttributeMapping);
+                    var sourceKey = setting.productAttributeSource;            
+                    var destKey = setting.productAttributeDestKey;
+                    return productList
+                        .filter(function (p) { return p && (p.Sku || p.Id); })
+                        .map(function (p) {
+                            var obj = {
+                                id: p.Sku || p.Id,
+                                quantity: isNumeric(p.Quantity) ? p.Quantity : 1,
+                                content_name: p.Name,
+                                name: p.Name,
+                                variant: p.Variant,
+                                category: p.Category
+                            };
+                            if (mappings) {

```

Comment:
`parseMappings` will create a `mappings` that always returns an array.  you can remove this if statement